full-stack-tests/backend/src/integration/SchemaSync.test.ts

```diff
@@ -0,0 +1,137 @@
+    await b2.pullChanges({ accessToken: user2AccessToken });
+    assert.isTrue(b2.nativeDb.schemaSyncEnabled());
+
+    // Import schema into b1 but do not push it.
+    const schema1 = `<?xml version="1.0" encoding="UTF-8"?>
+    <ECSchema schemaName="TestSchema1" alias="ts" version="01.00.00" xmlns="http://www.bentley.com/schemas/Bentley.ECXML.3.1">
+        <ECSchemaReference name="BisCore" version="01.00.00" alias="bis"/>
+        <ECEntityClass typeName="Pipe1">
+            <BaseClass>bis:GeometricElement2d</BaseClass>
+            <ECProperty propertyName="p1" typeName="int" />
+            <ECProperty propertyName="p2" typeName="int" />
+        </ECEntityClass>
+    </ECSchema>`;
+    await b1.importSchemaStrings([schema1]);
+    b1.saveChanges();
+
+    // ensure b1 have class and its properties
+    assert.sameOrderedMembers(["p1", "p2"], Object.getOwnPropertyNames(b1.getMetaData("TestSchema1:Pipe1").properties));
+
+    // pull schema change into b2 from shared schema channel
+    await synchronizeSchemas(b2);
+    b2.saveChanges();
+
+    // ensure b2 have class and its properties
+    assert.sameOrderedMembers(["p1", "p2"], Object.getOwnPropertyNames(b2.getMetaData("TestSchema1:Pipe1").properties));
+
+    // add new properties in b2
+    const schema2 = `<?xml version="1.0" encoding="UTF-8"?>
+    <ECSchema schemaName="TestSchema1" alias="ts" version="01.00.01" xmlns="http://www.bentley.com/schemas/Bentley.ECXML.3.1">
+        <ECSchemaReference name="BisCore" version="01.00.00" alias="bis"/>
+        <ECEntityClass typeName="Pipe1">
+          <BaseClass>bis:GeometricElement2d</BaseClass>
+          <ECProperty propertyName="p1" typeName="int" />
+          <ECProperty propertyName="p2" typeName="int" />
+          <ECProperty propertyName="p3" typeName="int" />
+          <ECProperty propertyName="p4" typeName="int" />
+        </ECEntityClass>
+    </ECSchema>`;
+    await b2.importSchemaStrings([schema2]);
+    b2.saveChanges();
+
+    // ensure b2 have class and its properties
+    assert.sameOrderedMembers(["p1", "p2", "p3", "p4"], Object.getOwnPropertyNames(b2.getMetaData("TestSchema1:Pipe1").properties));
+
+    // pull schema change into b1 from shared schema channel
+    await synchronizeSchemas(b1);
+    b1.saveChanges();
+
+    // ensure b1 have class and its properties
+    assert.sameOrderedMembers(["p1", "p2", "p3", "p4"], Object.getOwnPropertyNames(b1.getMetaData("TestSchema1:Pipe1").properties));
+
+    // push changes
+    await b1.pushChanges({ accessToken: user1AccessToken, description: "push schema changes" });
+    await b2.pushChanges({ accessToken: user2AccessToken, description: "push similar changes as b1" });
+
+    // b3 which has not seen any schema change
+    await b3.pullChanges({ accessToken: user3AccessToken });
+
+    // ensure b3 have class and its properties
+    assert.sameOrderedMembers(["p1", "p2", "p3", "p4"], Object.getOwnPropertyNames(b3.getMetaData("TestSchema1:Pipe1").properties));
+
```

Comment:
i don't see any test for errors (e.g. attempting to import the same schema from different briefcases).

Reply:
We have a native test for it. But i will add one here.